File: live/app/src/main/java/com/trtc/uikit/livekit/example/view/login/LoginActivity.java
package com.trtc.uikit.livekit.example.view.login;

import static com.tencent.qcloud.tuikit.debug.GenerateTestUserSig.TENCENT_EFFECT_LICENSE_KEY;
import static com.tencent.qcloud.tuikit.debug.GenerateTestUserSig.TENCENT_EFFECT_LICENSE_URL;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.util.Log;
import android.view.WindowManager;
import android.widget.EditText;

import com.tencent.imsdk.v2.V2TIMManager;
import com.tencent.imsdk.v2.V2TIMUserFullInfo;
import com.tencent.imsdk.v2.V2TIMValueCallback;
import com.tencent.qcloud.tuicore.TUILogin;
import com.tencent.qcloud.tuicore.interfaces.TUICallback;
import com.tencent.qcloud.tuicore.util.SPUtils;
import com.tencent.qcloud.tuicore.util.ToastUtil;
import com.tencent.qcloud.tuikit.debug.GenerateTestUserSig;
import com.trtc.uikit.livekit.common.Debug;
import com.trtc.uikit.livekit.example.BaseActivity;
import com.trtc.uikit.livekit.example.R;
import com.trtc.uikit.livekit.example.store.AppStore;
import com.trtc.uikit.livekit.example.view.main.MainActivity;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

public class LoginActivity extends BaseActivity {
    private static final String TAG = "LoginActivity";

    private EditText mEditUserId;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.app_activity_login);
        getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_HIDDEN);

        mEditUserId = findViewById(R.id.et_userId);
        mEditUserId.setText(SPUtils.getInstance("livekit").getString("userId"));
        findViewById(R.id.btn_login).setOnClickListener(v -> {
            String userId = mEditUserId.getText().toString().trim();
            SPUtils.getInstance("livekit").put("userId", userId);
            login(userId);
        });

        findViewById(R.id.iv_login_logo).setOnLongClickListener(v -> {
            if (!Debug.IS_DEBUG) {
                return false;
            }
            Debug.IS_TEST_ENV = !Debug.IS_TEST_ENV;
            ToastUtil.toastShortMessage("env:" + (Debug.IS_TEST_ENV ? "Test" : "Online"));
            return true;
        });
    }

    private void login(String userId) {
        if (TextUtils.isEmpty(userId)) {
            ToastUtil.toastShortMessage(getString(R.string.app_user_id_is_empty));
            return;
        }
        if (Debug.IS_TEST_ENV) {
            switchTestEnv();
        }
        TUILogin.login(this, GenerateTestUserSig.SDKAPPID, userId, GenerateTestUserSig.genTestUserSig(userId),
                new TUICallback() {
                    @Override
                    public void onSuccess() {
                        Log.i(TAG, "login success");
                        AppStore.userId = userId;
                        getUserInfo(userId);
                    }

                    @Override
                    public void onError(int errorCode, String errorMessage) {
                        ToastUtil.toastShortMessage(getString(R.string.app_toast_login_fail));
                        Log.e(TAG, "login fail，error code: " + errorCode + " message:" + errorMessage);
                    }
                });
    }

    private void switchTestEnv() {
        JSONObject jsonObject = new JSONObject();
        JSONObject params = new JSONObject();
        try {
            jsonObject.put("api", "setTestEnvironment");
            params.put("enableRoomTestEnv", true);
            jsonObject.put("params", params);

            V2TIMManager.getInstance().callExperimentalAPI("setTestEnvironment", true,
                    new V2TIMValueCallback<Object>() {
                        @Override
                        public void onSuccess(Object o) {
                            Log.i(TAG, "V2TIMManager setNetEnv success");
                        }

                        @Override
                        public void onError(int code, String desc) {
                            Log.i(TAG, "setNetEnv fail");
                        }
                    });
        } catch (JSONException e) {
            System.out.println("JSON error: " + e.getMessage());
        }
    }

    private void getUserInfo(String userId) {
        List<String> userList = new ArrayList<>();
        userList.add(userId);

        V2TIMManager v2TIMManager = V2TIMManager.getInstance();
        v2TIMManager.getUsersInfo(userList, new V2TIMValueCallback<List<V2TIMUserFullInfo>>() {
            @Override
            public void onError(int errorCode, String errorMsg) {
                Log.e(TAG, "getUserInfo failed, code:" + errorCode + " msg: " + errorMsg);
            }

            @Override
            public void onSuccess(List<V2TIMUserFullInfo> userFullInfoList) {
                if (userFullInfoList == null || userFullInfoList.isEmpty()) {
                    Log.w(TAG, "getUserInfo result is empty");
                    return;
                }
                V2TIMUserFullInfo timUserFullInfo = userFullInfoList.get(0);
                String userName = timUserFullInfo.getNickName();
                String userAvatar = timUserFullInfo.getFaceUrl();
                Log.d(TAG, "getUserInfo success: userName = " + userName + " , userAvatar = " + userAvatar);

                if (TextUtils.isEmpty(userName) || TextUtils.isEmpty(userAvatar)) {
                    startProfileActivity();
                } else {
                    AppStore.userAvatar = userAvatar;
                    AppStore.userName = userName;
                    startMainActivity();
                }
                finish();
            }
        });
    }

    private void startProfileActivity() {
        Intent intent = new Intent(LoginActivity.this, ProfileActivity.class);
        startActivity(intent);
    }

    private void startMainActivity() {
        Intent intent = new Intent(LoginActivity.this, MainActivity.class);
        startActivity(intent);
    }
}

File: application/app/src/main/java/com/tencent/uikit/app/login/LoginActivity.kt
package com.tencent.uikit.app.login

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.EditText
import android.widget.ImageView
import com.tencent.imsdk.v2.V2TIMManager
import com.tencent.imsdk.v2.V2TIMUserFullInfo
import com.tencent.imsdk.v2.V2TIMValueCallback
import com.tencent.qcloud.tuicore.TUILogin
import com.tencent.qcloud.tuicore.interfaces.TUICallback
import com.tencent.qcloud.tuicore.util.SPUtils
import com.tencent.qcloud.tuicore.util.ToastUtil
import com.tencent.qcloud.tuikit.debug.GenerateTestUserSig
import com.tencent.qcloud.tuikit.debug.GenerateTestUserSig.TENCENT_EFFECT_LICENSE_URL
import com.tencent.qcloud.tuikit.tuicallkit.TUICallKit.Companion.createInstance
import com.tencent.uikit.app.R
import com.tencent.uikit.app.main.BaseActivity
import com.tencent.uikit.app.main.MainActivity
import com.tencent.uikit.app.mine.UserManager

class LoginActivity : BaseActivity() {
    companion object {
        private const val TAG = "LoginActivity"
    }

    private var isTestEnv = false
    private lateinit var editUserId: EditText

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (!isTaskRoot
            && intent.hasCategory(Intent.CATEGORY_LAUNCHER)
            && intent.action != null
            && intent.action.equals(Intent.ACTION_MAIN)) {
            finish();
            return;
        }
        setContentView(R.layout.app_activity_login)
        initView()
    }

    private fun initView() {
        editUserId = findViewById(R.id.et_userId)
        editUserId.setText(SPUtils.getInstance("livekit").getString("userId"))
        findViewById<View>(R.id.btn_login).setOnClickListener {
            val userId = editUserId.text.toString().trim()
            SPUtils.getInstance("app_uikit").put("userId", userId)
            login(userId)
        }
        findViewById<ImageView>(R.id.iv_login_logo).setOnLongClickListener {
            isTestEnv = !isTestEnv
            switchTestEnv(isTestEnv)
            true
        }
    }

    private fun login(userId: String) {
        if (userId.isEmpty()) {
            ToastUtil.toastShortMessage(getString(R.string.app_user_id_is_empty))
            return
        }
        val userSig = GenerateTestUserSig.genTestUserSig(userId)
        TUILogin.login(this, GenerateTestUserSig.SDKAPPID, userId, userSig, object : TUICallback() {
            override fun onSuccess() {
                Log.i(TAG, "login onSuccess")
                val instance = createInstance(application)
                instance.enableFloatWindow(true)
                instance.enableVirtualBackground(true)
                instance.enableIncomingBanner(true)
                getUserInfo()
            }

            override fun onError(errorCode: Int, errorMessage: String?) {
                ToastUtil.toastShortMessage(
                    getString(R.string.app_toast_login_fail, errorCode, errorMessage)
                )
                Log.e(TAG, "login fail errorCode: $errorCode errorMessage:$errorMessage")
            }
        })
    }

    private fun getUserInfo() {
        UserManager.getInstance().getSelfUserInfo(object : V2TIMValueCallback<V2TIMUserFullInfo> {
            override fun onError(errorCode: Int, errorMsg: String?) {
                Log.e(TAG, "getUserInfo failed, code:$errorCode msg: $errorMsg")
            }

            override fun onSuccess(timUserFullInfo: V2TIMUserFullInfo?) {
                if (timUserFullInfo == null) {
                    Log.e(TAG, "getUserInfo result is empty")
                    return
                }
                val userName = timUserFullInfo.nickName
                val userAvatar = timUserFullInfo.faceUrl
                if (userName.isNullOrEmpty() || userAvatar.isNullOrEmpty()) {
                    val intent = Intent(this@LoginActivity, RegisterActivity::class.java)
                    startActivity(intent)
                    finish()
                } else {
                    val intent = Intent(this@LoginActivity, MainActivity::class.java)
                    startActivity(intent)
                    finish()
                }
            }
        })
    }

    private fun switchTestEnv(enableTestEnv: Boolean) {
        V2TIMManager.getInstance().callExperimentalAPI(
            "setTestEnvironment", enableTestEnv, object : V2TIMValueCallback<Any?> {
                override fun onSuccess(o: Any?) {
                    Log.i(TAG, "V2TIMManager setNetEnv success. enableTestEnv：$enableTestEnv")
                }

                override fun onError(code: Int, desc: String?) {
                    Log.i(TAG, "V2TIMManager setNetEnv fail. enableTestEnv：$enableTestEnv")
                }
            })
        if (enableTestEnv) {
            ToastUtil.toastShortMessage("Test Env")
        } else {
            ToastUtil.toastShortMessage("Online Env")
        }
    }
}